tasks.py

```diff
@@ -0,0 +1,198 @@
+import os
+import json
+import git
+import sys
+import cfenv
+
+from invoke import task
+
+env = cfenv.AppEnv()
+
+APP_NAME = "fecfile-image-generator"
+ORG_NAME = "fec-fecfileonline-prototyping"
+
+
+def _resolve_rule(repo, branch):
+    """Get space associated with first matching rule."""
+    for space, rule in DEPLOY_RULES:
+        if rule(repo, branch):
+            return space
+    return None
+
+
+def _detect_branch(repo):
+    try:
+        return repo.active_branch.name
+    except TypeError:
+        return None
+
+
+def _detect_space(repo, branch=None):
+    """Detect space from active git branch.
+
+    :param str branch: Optional branch name override
+    :returns: Space name if space is detected and confirmed, else `None`
+    """
+    space = _resolve_rule(repo, branch)
+    if space is None:
+        print("The current configuration does not require a deployment to cloud.gov.")
+        return None
+    print(f"Detected space {space}")
+    return space
+
+
+DEPLOY_RULES = (
+    ("prod", lambda _, branch: branch == "main"),
+    ("stage", lambda _, branch: branch.startswith("release")),
+    ("dev", lambda _, branch: branch == "dev"),
```

Comment:
I think this should be branch == "develop" rather than dev so that it matches our default development branch name of "develop". Does that sound right to you?